src/main/integrations/sentry-minidump/minidump-parser.ts

```diff
@@ -0,0 +1,256 @@
+  // const _reserved = buf.readUInt16LE(offset + 6)
+  const value = buf.readUInt32LE(offset + 8);
+
+  if (ty === 1) {
+    return [readStringUtf8Unterminated(all, name), readStringUtf8Unterminated(all, value)];
+  }
+
+  return undefined;
+}
+
+function readAnnotationObjects(buf: Buffer, location: Location): Record<string, string> {
+  const data = buf.subarray(location.rva, location.rva + location.dataSize);
+  if (data.length === 0) {
+    return {};
+  }
+
+  const annotationObjectsLocation = readCrashpadModuleInfoAnnotationObjectsLocation(data, 0);
+  const annotationObjectsData = buf.subarray(
+    annotationObjectsLocation.rva,
+    annotationObjectsLocation.rva + annotationObjectsLocation.dataSize,
+  );
+
+  const count = annotationObjectsData.readUInt32LE(0);
+  let offset = 4;
+
+  const annotationObjects: Record<string, string> = {};
+  for (let i = 0; i < count; i++) {
+    const annotation = readAnnotationObject(annotationObjectsData, buf, offset);
+    if (annotation) {
+      const [key, value] = annotation;
+      annotationObjects[key] = value;
+    }
+    offset += 12;
+  }
+
+  return annotationObjects;
+}
+
+function readCrashpadModuleLinks(buf: Buffer, location: Location): Record<string, string> {
+  const data = buf.subarray(location.rva, location.rva + location.dataSize);
+
+  if (data.length === 0) {
+    return {};
+  }
+
+  const count = data.readUInt32LE(0);
+  let offset = 4;
+
+  let annotationObjects = {};
+  for (let i = 0; i < count; i++) {
+    const annotationObjectsLocation = readLocationDescriptor(data, offset + 4);
+    annotationObjects = { ...annotationObjects, ...readAnnotationObjects(buf, annotationObjectsLocation) };
+    offset += 12;
+  }
+
+  return annotationObjects;
+}
+
+// https://crashpad.chromium.org/doxygen/structcrashpad_1_1MinidumpCrashpadInfo.html
+function parseCrashpadInfo(buf: Buffer, info: Buffer): Record<string, string> {
+  // This stream is of the following format:
+  // pub struct MINIDUMP_CRASHPAD_INFO {
+  //   pub version: u32,
+  //   pub report_id: GUID,
+  //   pub client_id: GUID,
+  //   pub simple_annotations: MINIDUMP_LOCATION_DESCRIPTOR,
+  //   pub module_list: MINIDUMP_LOCATION_DESCRIPTOR,
+  // }
+
+  const module_list = readLocationDescriptor(info, 44);
+
+  return readCrashpadModuleLinks(buf, module_list);
+}
+
+type CrashpadAnnotations = {
+  process_type?: string;
+} & Record<string, string>;
+
+export type MinidumpParseResult = {
+  header: MinidumpHeader;
+  crashpadAnnotations?: CrashpadAnnotations;
+};
+
+/** Parses an Electron minidump and extracts the header and crashpad annotations */
+export function parseMinidump(buf: Buffer): MinidumpParseResult | undefined {
+  if (buf.length < 32) {
+    return undefined;
+  }
+
+  let header: MinidumpHeader | undefined;
+  try {
+    header = readHeader(buf);
+  } catch (_) {
+    return undefined;
+  }
+
+  if (header.signature !== MINIDUMP_MAGIC_SIGNATURE || buf.length < 10_000) {
```

Comment:
Why the check on 10000?

Reply:
This check existed previously to exclude invalid looking minidumps as reported by users. I'll find the issue and add a link in a comment.